SharedPackages/DataBrokerProtectionCore/Tests/DataBrokerProtectionCoreTests/BrokerProfileQueryDataTests.swift

```diff
@@ -0,0 +1,136 @@
+        let queryData2 = BrokerProfileQueryData.mock(dataBrokerName: "2", lastRunDate: Date(timeIntervalSince1970: 1000))
+        let queryData3 = BrokerProfileQueryData.mock(dataBrokerName: "3", lastRunDate: Date(timeIntervalSince1970: 500))
+        let array = [queryData1, queryData2, queryData3]
+
+        let result = array.elementsSortedByScanLastRunDateWhereScansRanBetween(earlierDate: Date(timeIntervalSince1970: 400), laterDate: Date(timeIntervalSince1970: 20000))
+        XCTAssertEqual(result.map { $0.dataBroker.name }, ["3", "2"])
+    }
+
+    // elementsSortedByScanPreferredRunDateWhereDateIsBetween tests
+
+    func test_elementsSortedByScanPreferredRunDateWhereDateIsBetween_whenNoDatesBetweenDates_ThenReturnsEmptyArray() {
+        let queryData1 = BrokerProfileQueryData.mock(dataBrokerName: "1", preferredRunDate: Date(timeIntervalSince1970: 0))
+        let queryData2 = BrokerProfileQueryData.mock(dataBrokerName: "2", preferredRunDate: Date(timeIntervalSince1970: 1000))
+        let queryData3 = BrokerProfileQueryData.mock(dataBrokerName: "3", preferredRunDate: Date(timeIntervalSince1970: 500))
+        let array = [queryData1, queryData2, queryData3]
+
+        let result = array.elementsSortedByScanPreferredRunDateWhereDateIsBetween(earlierDate: Date(timeIntervalSince1970: 2000), laterDate: Date(timeIntervalSince1970: 20000))
+        XCTAssertEqual(result.count, 0)
+    }
+
+    func test_elementsSortedByScanPreferredRunDateWhereDateIsBetween_whenSomeDatesBetweenDates_ThenReturnsSortedArray() {
+        let queryData1 = BrokerProfileQueryData.mock(dataBrokerName: "1", preferredRunDate: Date(timeIntervalSince1970: 0))
+        let queryData2 = BrokerProfileQueryData.mock(dataBrokerName: "2", preferredRunDate: Date(timeIntervalSince1970: 1000))
+        let queryData3 = BrokerProfileQueryData.mock(dataBrokerName: "3", preferredRunDate: Date(timeIntervalSince1970: 500))
+        let array = [queryData1, queryData2, queryData3]
+
+        let result = array.elementsSortedByScanPreferredRunDateWhereDateIsBetween(earlierDate: Date(timeIntervalSince1970: 400), laterDate: Date(timeIntervalSince1970: 20000))
+        XCTAssertEqual(result.map { $0.dataBroker.name }, ["3", "2"])
+    }
+
+    /*
+
+
+     func elementsSortedByScanPreferredRunDateWhereDateIsBetween(earlierDate: Date, laterDate: Date) -> [BrokerProfileQueryData] {
+         guard earlierDate < laterDate else {
+             assertionFailure()
+             return []
+         }
+
+         let unsortedElementsBetweenDates = self.filter {
+             $0.scanJobData.preferredRunDate != nil &&
+             $0.scanJobData.preferredRunDate! >= earlierDate &&
+             $0.scanJobData.preferredRunDate! <= laterDate
+         }
+
+         let sortedElements = unsortedElementsBetweenDates.sorted {
+             $0.scanJobData.preferredRunDate! < $1.scanJobData.preferredRunDate!
+         }
+
+         return sortedElements
+     }
+     */
+}
```

Comment:
Duplicate, uncommented

Reply:
Oops, I like to look at the function I'm testing when writing the tests 😅